diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 3d23b318f..4f7779356 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1825,6 +1825,11 @@ config("exceptions") {
   } else {
     cflags_cc = [ "-fexceptions" ]
     cflags_objcc = cflags_cc
+    if (is_mingw && exclude_unwind_tables) {
+      # With exceptions explicitly allowed, override exclude_unwind_tables or
+      # linking will fail on MinGW
+      cflags_cc += [ "-funwind-tables", "-fasynchronous-unwind-tables" ]
+    }
   }
 }
 
